src/jsifier.mjs

```diff
@@ -79,6 +80,17 @@ function stringifyWithFunctions(obj) {
   if (Array.isArray(obj)) {
     return '[' + obj.map(stringifyWithFunctions).join(',') + ']';
   }
+
+  // preserve the type of the object if it is one of [Map, Set, WeakMap, WeakSet].
+  const builtinContainers = vm.runInContext('[Map, Set, WeakMap, WeakSet]', compileTimeContext);
```

Comment:
I think we have a helper called runinmacrocontext or something like that, which would avoid the direct use of VM here

Reply:
There is a function `runInMacroContext`, which requires a `filename` in options. I don't think that function works here.

do you want me to add the `runInContext` into utility.mjs like this:
```js
export function runInContext(code) {
  return vm.runInContext(code, compileTimeContext);
}
```

Reply:
I think you can just modify `runInMacroContext` such that the filename is optional, or (perhaps better) pass `filename: '<internal>'`.

Reply:
Also, can you look into doing this instead:

```
export const compileTimeContext = vm.createContext(vm.constants.DONT_CONTEXTIFY);
addToCompileTimeContext({process, console})
```

Then I think you can simply access `compileTimeContext.Set` .. without running any code.